test_runner/regress/test_storage_controller.py

```diff
@@ -2862,6 +2862,143 @@ def new_becomes_leader():
         )
 
 
+def test_storage_controller_leadership_transfer_during_split(
```

Comment:
At the risk of being pedantic: what's the success condition for this test? The absence of error log lines?
Would be nice to add something at the end that checks the tenant shards are present and available.

Reply:
It's mainly the consistency_check() call -- that detects if what's in the database is bogus.  But yeah, it could be strengthened